epub33/core/index.html

```diff
@@ -1213,6 +1213,29 @@ <h3>Exempt resources</h3>
 							CSS style sheets) is an exempt resource.</p>
 					</dd>
 
+					<div class="proposed correction" id="change_1">
+						<span class="marker">Proposed Correction 1</span>
+						<p>The exemption for resources not referenced from the spine or embedded in content documents
+							(refer to the paragraphs following note below), includes resources referenced from
```

Comment:
This isn't really true - resources referenced from the script element itself did require a fallback before this change, so we are in fact loosening the restriction here. I am not sure if that still falls in class 3, maybe it does? But it isn't really a required change at this time, wasm is not technically importable via the script element (yet). Happy leaving this if we do decide this is still class 3.

Reply:
> resources referenced from the script element itself did require a fallback before this change

I'd argue not this change but the change that added the exemption below. It was probably too broad in hindsight, as the script element doesn't render the text of its script (it's not embedding the script in the content document; it's there to execute). So if a script is not embedded content and not referenced from the spine, then it's exempt.

If we don't include this exemption explicitly, the only way to make script require core media types again is to do a different class 3 change that adds it along with the embeddable content elements. I'd rather just move ahead this way.

Reply:
I guess I am getting lost in the reviews here. Which is the change that added the exemption? At least in the github UI I see "Any resource referenced from the html script element that is not already a core media type resource [snip] is an exempt resource." That seems to be a newly added change. Are you saying that it was already implied because referencing from the script element isn't really embedding? It's an interesting argument, and hard to dispute since we don't really define what it means to embed something.

Reply:
> It's an interesting argument, and hard to dispute since we don't really define what it means to embed something.

Right, in 3.3 we added the new exemption for data files and such. The first condition is the resource not be in the spine and the second that:

> it is not embedded directly in EPUB content documents (e.g., via [[html](https://www.w3.org/TR/epub-33/#bib-html)] [embedded content](https://html.spec.whatwg.org/multipage/dom.html#embedded-content-category) and [[svg](https://www.w3.org/TR/epub-33/#bib-svg)] [image](https://www.w3.org/TR/SVG/embedded.html#ImageElement) and [foreignObject](https://www.w3.org/TR/SVG/embedded.html#ForeignObjectElement) elements).

Embedding has always been the term we use for rendering a resource within a content document (via `object`, `img`, `iframe`, etc.). We usually use some form of "referenced from" if we mean any kind of reference (`script`, `link`, `a`, etc.).

`script` and `link` both allow references to resources without embedding them. That's partly why we made explicit that resources referenced from `link` are exempt in 3.3 (plus there's a stronger case that nothing uses them than with script).

It's also why, at least to me, we've already (accidentally) opened the door to the `script` element not needing backups in the current spec. If we opt not to keep it exempt, we'd need to make that second bullet clearer.

Another reason it probably makes more sense to exempt `script` is that the only way you can provide an alternative for a resource referenced from it is through a manifest fallback. The less of that we have the better. :smile:

Reply:
I just did a review of embed use in the spec, and I can't say the use is as clear as you make it sound, It is certainly never defined that way. We do talk about embedded scripts (i.e scripts that are written directly in a script element), so those scripts are embedded, but if you move it to another file they are referenced? But we don't have that distinction for images (a data url embedded image does not become referenced when we move it to a new file). And what about CSS? If it is in another file is it referenced and therefore exempt? 

FWIW, my understanding of embedded vs referenced is embedded content is used by the current document in the (continuing) process of rendering it, while referenced content does not participate in the rendering of the current document, it is simply mentioned (links, etc). Of course, that isn't defined anywhere either.

Reply:
> It is certainly never defined that way.

No, that's a big part of the problem, but that might be a useful cleanup step for a future update.

But in this case we've selectively picked definitions and elements that are for rendering a resource directly. I can't square that with how script works.

Reply:
Are you arguing against having the exemption for script, though, or is this just about whether the resources were exempt before or not? (I'll get on editing out some of that prose now, since it's arguable both ways.)

I don't see how making script resources exempt would go up to a class 4 change, for example, even if we disagree over the current exemptions. Exempt resources are not new and exempting resources only gets them out of fallbacks. It doesn't add a new feature to the specification.